editor/src/messages/portfolio/document/node_graph/node_graph_message_handler.rs

```diff
@@ -397,12 +412,26 @@ impl NodeGraphMessageHandler {
 					warn!("Removing input of invalid node type '{}'", node.name);
 					return false;
 				};
+
 				if let NodeInput::Value { tagged_value, .. } = &node_type.inputs[input_index].default {
-					*input = NodeInput::value(tagged_value.clone(), true);
+					let mut refers_to_output_node = false;
+
+					if let Some(input_node) = &first_input_node {
+						if let Some(output_node) = first_output_node {
+							if output_node.node_id == *node_id {
+								refers_to_output_node = true;
+								*input = input_node.clone()
+							}
+						}
+					}
```

Comment:
Let's say you have the following network and delete the greyscale node.
![image](https://github.com/GraphiteEditor/Graphite/assets/78500760/5391d949-0742-4e0f-a2fd-6ee5c7db91e7)
We find that the invert node uses the (now deleted) greyscale node as input. We need to modify the input node so that instead refers to something else.

Once we reach this code block, we know that we are dealing with a node with an input that used the deleted node (i.e. the invert node's first input).

We must also check if we are using the primary output from the now deleted greyscale node. You can do this by modifying the above code:
```rs
let NodeInput::Node { node_id, .. } = input else {
	continue;
};
```
to be
```rs
let NodeInput::Node { node_id, output_index, .. } = input else {
	continue;
};
```
Then you can check `if output_index == 0`.

Previously we just reset it to default. Now we need to reset it to the `first_input_node`, if that exists. If it doesn't then we should use the old behaviour `*input = NodeInput::value(tagged_value.clone(), true);`.